src/App.tsx

```diff
@@ -1,80 +1,61 @@
 import Header from './components/Header';
 import Login from './components/Login';
 import './App.css';
-import { useContext, useEffect, useState } from 'react';
-import { ExceptionType } from '@perspect3vism/ad4m';
-import { ExceptionInfo } from '@perspect3vism/ad4m/lib/src/runtime/RuntimeResolver';
-import { Loader, Stack } from '@mantine/core';
-import { Ad4mContext } from '.';
+import { useContext, useState } from 'react';
+import { Button, Loader, Stack, TextInput } from '@mantine/core';
 import TrustAgent from './components/TrustAgent';
 import Navigation from './components/Navigation';
+import { Ad4minContext } from './context/Ad4minContext';
 
 const App = () => {
-  const ad4mClient = useContext(Ad4mContext);
+  const {state: {
+    connected, isUnlocked, candidate, connectedLaoding, did
+  }, methods: {
+    handleTrustAgent,
+    setUrl
```

Comment:
This is pretty confusing `setUrl` with below `setURL`.
probably change this one to `configureEndpoint`?